sink/src/main/java/com/spredfast/kafka/connect/s3/sink/S3SinkTask.java

```diff
@@ -102,6 +103,32 @@ public void stop() throws ConnectException {
 		}
 	}
 
+	@Override
+	public Map<TopicPartition, OffsetAndMetadata> preCommit(
+		Map<TopicPartition, OffsetAndMetadata> currentOffsets) {
+		Map<TopicPartition, OffsetAndMetadata> result = new HashMap<>();
+
+		log.info("{} performing preCommit", name());
+		log.info("{} offsets: {}", name(), currentOffsets);
+		log.info("{} compressed_file_size: {}", name(), GZIPThreshold);
+
+		partitions.entrySet().stream()
+			.filter(e -> e.getValue() != null && e.getValue().getWriter() != null && e.getValue().getWriter().getTotalCompressedSize() > GZIPThreshold)
+			.forEach(e -> {
+				BlockGZIPFileWriter blockWriter = e.getValue().getWriter();
+				log.info("{} partition: {}, totalCompressedSize: {}", name(), e.getValue(), blockWriter.getTotalCompressedSize());
+				result.put(e.getKey(), currentOffsets.get(e.getKey()));
+			});
+
+		if (!result.isEmpty()) {
+			log.info("{} result: {}", name(), result);
+			flush(result);
+			flushedOffsets.putAll(result);
+		}
+
+		return flushedOffsets;
+	}
+
```

Comment:
Log messages were injected just for some easier debugging. They need to be cleaned up.